pkg/log/logger.go

```diff
@@ -87,3 +91,10 @@ func (l Logger) V(level int) Logger {
 	l.Logger = l.Logger.V(level)
 	return l
 }
+
+func (l Logger) SetMaxVerbosity(v int) {
+	c, ok := l.Logger.GetSink().(ConfigurableLogSink)
+	if ok {
+		c.SetMaxVerbosity(v)
+	}
+}
```

Comment:
Since `ConfigurableLogSink` isn’t used anywhere else (and shouldn’t be, AFAICT), how about

```suggestion
func (l Logger) SetMaxVerbosity(v int) {
	type configurableLogSink interface {
		SetMaxVerbosity(v int)
	}

	c, ok := l.Logger.GetSink().(configurableLogSink)
	if ok {
		c.SetMaxVerbosity(v)
	}
}
```

Reply:
It's public so it's visible to implementors, currently only `zeroLogContext`. I changed it to `configurableLogSink` and the compiler still recognizes `zeroLogContext` as an implementation but it seems odd.

Reply:
That’s perfectly fine — Go is close to duck typing. Go interfaces aren’t so much an implementation contract as they are a _type usage_ contract, and in many cases they’re relevant for code _using_ types, not implementing them. See https://go.dev/wiki/CodeReviewComments#interfaces for background. In this instance, the interface seems like a proxy for a method presence check — that is to say, I understand the intent of the code to be “if `l.logger.GetSink()` has a `SetMaxVerbosity` method taking an `int` parameter, call it. The way to represent that in Go is with a private, usage-side interface.

Reply:
OK.  I guess the benefit to having it public is to ensure the desired interface is implemented, ie

`var _ log.ConfigurableLogSink = &zeroLogContext{}`

But I'm fine either way - I don't think it really matters.